addons/web/static/src/core/dialog/dialog.scss

```diff
@@ -4,4 +4,13 @@
     button {
         margin: 0; // Reset boostrap.
     }
+
+    footer {
+        > :not(:first-child) {
+          margin-left: 0.25rem;
+        }
+        > :not(:last-child) {
+          margin-right: 0.25rem;
+        }
```

Comment:
I'm not sure but why not just adding a margin left if the element is not the first one? 🤔 
Also, why not select only the buttons?

Reply:
@xavierbol , done with the margin-left. As this is a generic fix so there might be chances that the footer may contain a link(a tag) or icon and we also want to add some space on that too so applied for a generic footer instead of a button.  Thanks